File: ts/proof/verify.ts
import crypto from "crypto";

export function verifyAttestation(
  payload: string,
  signature: string,
  publicKey: string
): boolean {
  const verify = crypto.createVerify("SHA256");
  verify.update(payload);
  verify.end();
  return verify.verify(publicKey, signature, "base64");
}

File: py/proof/verify.py
from cryptography.hazmat.primitives import serialization, hashes
from cryptography.hazmat.primitives.asymmetric import padding

def verify_attestation(payload: bytes, signature: bytes, public_key_pem: bytes):
    pub = serialization.load_pem_public_key(public_key_pem)
    pub.verify(
        signature,
        payload,
        padding.PKCS1v15(),
        hashes.SHA256()
    )
    return True